device.yml

```diff
@@ -0,0 +1,273 @@
+        description: Specifies the polarity of the trigger signal.
+        mask: 0x10
+    description: Trigger mode of input channel 0.
+  Trigger1Mode:
+    <<: *triggermode
+    address: 65
+    description: Trigger mode of input channel 1.
+  Trigger2Mode:
+    <<: *triggermode
+    address: 66
+    description: Trigger mode of input channel 2.
+  Trigger3Mode:
+    <<: *triggermode
+    address: 67
+    description: Trigger mode of input channel 3.
+  RequestEnable:
+    address: 68
+    type: U8
+    access: Write
+    maskType: PwmChannels
+    description: If set to Low, it will bypass the EnablePwmChannels and ArmPwmChannels register function.
+  EnablePwmChannels:
+    address: 69
+    type: U8
+    access: Write
+    maskType: PwmChannels
+    description: Enable enable
+  TriggerAllMode:
+    address: 70
+    type: U8
+    access: Write
+    payloadSpec:
+      TriggerMode:
+        maskType: TriggerAllModeConfig
+        description: Specifies the trigger mode.
+        mask: 0x03
+      Polarity:
+        maskType: TriggerPolarity
+        description: Specifies the polarity of the trigger signal.
+        mask: 0x10
+    description: When triggered, all channels will be affected by the event.
+  TriggerChannelState:
+    address: 71
+    type: U8
+    access: Read
+    maskType: TriggerInput
+    description: Current state of all trigger channel inputs.
+  PwmChannelState:
+    address: 72
+    type: U8
+    access: Read
+    maskType: PwmChannels
+    description: Current state of all PWM channel outputs.
+  PwmState:
+    address: 73
+    type: U8
+    access: Event
+    maskType: PwmChannels
+    description: The state of the PWM for each channel. Emits an event each time a PMW starts or stops.
+  EnableEvents:
+    address: 74
+    type: U8
+    access: Write
+    maskType: MultiPwmEvents
+    description: Enables the generation of events.
+bitMasks:
+  PwmChannels:
+    description: Available PWM output channels.
+    bits:
+      Channel0: 0x01
+      Channel1: 0x02
+      Channel2: 0x04
+      Channel3: 0x08
+  TriggerInput:
+    description: Available trigger input channels.
+    bits:
+      Channel0: 0x01
+      Channel1: 0x02
+      Channel2: 0x04
+      Channel3: 0x08
+  MultiPwmEvents:
+    description: Available events that can be enabled/disabled.
+    bits:
+      Execution: 0x01
+groupMasks:
+  PlaybackMode:
+    description: Available playback modes of the PWM channel.
+    values:
+      Count: 0
+      Infinite: 1
+  TriggerModeConfig:
+    description: Available operation modes for trigger input channels.
+    values:
+      Start: 0
+      StartAndStop: 1
+  TriggerPolarity:
+    description: Available polarity options for a trigger mode.
+    values:
+      Default: 0
+      Inverted: 8
+  TriggerAllModeConfig:
+    description: Available operation modes for "All" trigger input channel.
+    values:
+      Start: 0
+      StartAndStop: 1
+      Enable: 2
+      EnableAndStop: 3
```

Comment:
Why is this enum different from `TriggerModeConfig`? It feels strange to have more options when configuring all channels at the same time versus one at a time, but maybe I am missing something here since I don't know the device.

Reply:
The additional options for `TriggerModeConfig` are not directly related to the Trigger Channels. The additional `Enable` options allow to enable the PWM channels and not the Trigger Channels. If I remember it was added here for convenience, since we could use this "All" channel to enable by an external signal all the PWM Channels.

Reply:
I see. In that case probably the name `TriggerAllModeConfig` is misleading since it provides no indication that different channel types other than trigger may also be configured. I can see two options to improve the situation:

1. Keep the register name but remove the convenience functionality and make this register work only with the `TriggerModeConfig` group mask (would require firmware changes);
2. Change the register name to encompass both trigger and PWM channels, e.g. `TriggerAndPwmMode` or `AllChannelMode`. Also rename the group mask to match the register name with a `Config` suffix and rename the values in this group mask to include information about which channels are affected, e.g. `TriggerStart`, `TriggerStartAndStop`, `PwmEnable`, `PwmEnableAndStop`.

I would be happy with either choice, depending on how this register is meant to be used.

One last thing which is still not clear to me in this combined register is how the register can work with multiple channel types and be a group mask at the same time.

What happens if you write `TriggerStart` and then `PwmEnable`? Are these options mutually exclusive? Or will it leave the trigger mode at whatever value was set previously? In the latter case, what happens when you read the register?

Depending on the answers we could also consider the following options:

3. Convert the register type into a bitmask;
4. Make a separate register for all PWM channel config.

Both would also likely require firmware changes.

Reply:
I would prefer to improve the naming than making any changes to the firmware at this point. So I will modify the register, group masks name and values according to 2. and regenerate the interface if you have no additional suggestions.

These options are mutually exclusive.